plonky2x/src/frontend/hash/sha/sha256_curta.rs

```diff
@@ -2,46 +2,183 @@ use curta::chip::hash::sha::sha256::builder_gadget::{SHA256Builder, SHA256Builde
 use curta::chip::hash::sha::sha256::generator::SHA256HintGenerator;
 use curta::math::field::Field;
 use itertools::Itertools;
+use plonky2::iop::target::Target;
 use plonky2::plonk::config::{AlgebraicHasher, GenericConfig};
 
 use crate::backend::circuit::PlonkParameters;
 use crate::frontend::hash::bit_operations::util::u64_to_bits;
+use crate::frontend::uint::uint32::U32Variable;
 use crate::frontend::vars::Bytes32Variable;
-use crate::prelude::{ByteVariable, CircuitBuilder, CircuitVariable};
+use crate::prelude::{BoolVariable, ByteVariable, CircuitBuilder, CircuitVariable};
 
```

Comment:
1) move the helper functions into the builder
2) namespace the functions such that its `curta_<fn_name>` similar to the beacon/storage subfolders